--- a/hollywood/app/about_dialog.c
+++ b/hollywood/app/about_dialog.c
@@ -170,7 +170,7 @@ about_dialog_create (int timeout)
 #else
       about_dialog = gtk_window_new (GTK_WINDOW_DIALOG);
 	about_dialog->name = "about dialog";
-	  g_return_if_fail(!about_dialog);
+	  g_return_if_fail(about_dialog != NULL);
 #endif
       minimize_register(about_dialog);
       gtk_window_set_wmclass (GTK_WINDOW (about_dialog), "about_dialog", "cinepaint");
