diff --git a/src/viola/cl_generic.c b/src/viola/cl_generic.c
--- a/src/viola/cl_generic.c
+++ b/src/viola/cl_generic.c
@@ -57,6 +57,7 @@
 #include <stdlib.h>
 #include <sys/file.h>
 #include <unistd.h>
+#include <time.h>
 
 #include "../libWWW/HTParse.h"
 #include "../libWWW/HTTP.h"
@@ -1240,6 +1241,7 @@
     packet2->info.s = saveString("");
     packet3->info.s = saveString(path);
 lameLoopEsc:
+;
 
     char* anchor = HTParse(addr, relative, PARSE_ANCHOR);
     packet4->info.s = (anchor ? saveString(anchor) : saveString(""));
